dbt/main.py

```diff
@@ -92,7 +93,10 @@ def main(args=None):
 
         if logger_initialized:
             logger.debug(traceback.format_exc())
-        else:
+        elif not isinstance(e, RuntimeException):
```

Comment:
Should this be `RuntimeError`? It looks like we raise `RuntimeError` directly in a bunch of places. Maybe we should wrap those as `RuntimeException`s instead?

In practice, messing with my dbt project results in:
```
Encountered an error while reading the project:
Runtime Error
  Runtime Error
    Invalid arguments passed to "Project" instance: 'name' is a required property
Defined profiles:
 - default
 - bq
 - user
 - integration_tests

For more information on configuring profiles, please consult the dbt docs:

https://docs.getdbt.com/docs/configure-your-profile

Encountered an error:
Could not run dbt
Traceback (most recent call last):
  File "/Users/drew/fishtown/dbt/dbt/main.py", line 76, in main
    results, succeeded = handle_and_check(args)
  File "/Users/drew/fishtown/dbt/dbt/main.py", line 127, in handle_and_check
    task, res = run_from_args(parsed)
  File "/Users/drew/fishtown/dbt/dbt/main.py", line 168, in run_from_args
    raise RuntimeError("Could not run dbt")
RuntimeError: Could not run dbt
```

I think the direct cause of this is on line 168 of main.py

Reply:
I'll add: This particular error message is particular junky, and we probably shouldn't be showing a list of profiles for failed `dbt_project.yml` validation, but we can possibly tackle that separately...

Reply:
That sounds good, though I think changing those `raise RuntimeError` to `raise RuntimeException` is probably the better path.

I agree it's a bad error message, but it is a generic situation. Maybe we should just return there instead of raising?

Reply:
Yeah, this is good for now. We can overhaul error handling for config stuff in the future. There's a lot to do there, and it's not super high priority at the moment